include/nbl/asset/IDescriptorSetLayout.h

```diff
@@ -359,34 +361,42 @@ class IDescriptorSetLayout : public virtual core::IReferenceCounted  // TODO: tr
 
 		for (const auto& b : _bindings)
 		{
-			buildInfo_descriptors[static_cast<uint32_t>(b.type)].emplace_back(b.binding, b.createFlags, b.stageFlags, b.count);
-
-			if (b.type == IDescriptor::E_TYPE::ET_COMBINED_IMAGE_SAMPLER)
-			{
-				if (b.samplers)
-					buildInfo_immutableSamplers.emplace_back(b.binding, b.createFlags, b.stageFlags, b.count);
-				else
-					buildInfo_mutableSamplers.emplace_back(b.binding, b.createFlags, b.stageFlags, b.count);
+			switch (b.type) {
+				case IDescriptor::E_TYPE::ET_SAMPLER:
+					if (b.samplers)
+						buildInfo_immutableSamplers.emplace_back(b.binding, b.createFlags, b.stageFlags, b.count);
+					else 
+						buildInfo_descriptors[static_cast<uint32_t>(b.type)].emplace_back(b.binding, b.createFlags, b.stageFlags, b.count);
```

Comment:
you already know the type, so use that instead of `static_cast<uint32_t>(b.type)`